tests/ui/needless_borrows_for_generic_args.rs

```diff
@@ -333,4 +333,12 @@ fn main() {
         f(&y); // Don't lint
         f(&"".to_owned()); // Lint
     }
+    {
+        fn takes_writer<T: std::io::Write>(_: T) {}
+
+        fn f(mut buffer: &mut Vec<u8>) {
```

Comment:
NIT: I think the example in itself doesn't explain the issue well. This can easily be fixed by adding a comment or renaming the function to mention the issue:

```suggestion
        fn issue_12856(mut buffer: &mut Vec<u8>) {
```